src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -338,23 +451,25 @@ default MemorySegment allocateArray(MemoryLayout elementLayout, long count) {
     }
 
     /**
-     * Allocates a memory segment with the given size.
-     * @implSpec the default implementation for this method calls {@code this.allocate(byteSize, 1)}.
+     * {@return a new memory segment with the given {@code byteSize}}
+     *
+     * @implSpec The default implementation for this method calls
+     * {@code this.allocate(byteSize, 1)}.
+     *
      * @param byteSize the size (in bytes) of the block of memory to be allocated.
-     * @return a segment for the newly allocated memory block.
      * @throws IllegalArgumentException if {@code byteSize < 0}
      */
     default MemorySegment allocate(long byteSize) {
         return allocate(byteSize, 1);
     }
 
     /**
-     * Allocates a memory segment with the given size and alignment constraint.
+     * {@return a new memory segment with the given {@code byteSize} and {@code byteAlignment}}
+     *
      * @param byteSize the size (in bytes) of the block of memory to be allocated.
      * @param byteAlignment the alignment (in bytes) of the block of memory to be allocated.
-     * @return a segment for the newly allocated memory block.
      * @throws IllegalArgumentException if {@code byteSize < 0}, {@code byteAlignment <= 0},
-     * or if {@code byteAlignment} is not a power of 2.
+     *                                  or if {@code byteAlignment} is not a power of 2.
```

Comment:
I'd leave this out. Not because I disagree stylistically, but simply because it's a style that we don't adhere with in this package. If we want to change that, it's ok, but perhaps better to deal with it in another PR.